rp/src/key.rs

```diff
@@ -0,0 +1,105 @@
+use std::{fs, path::Path};
+
+use anyhow::{anyhow, Result};
+use wireguard_keys::Privkey;
+
+use rosenpass::protocol::{SPk, SSk};
+use rosenpass_cipher_traits::Kem;
+use rosenpass_ciphers::kem::StaticKem;
+use rosenpass_secret_memory::file::StoreSecret as _;
+
+pub fn genkey(private_keys_dir: &Path) -> Result<()> {
+    if private_keys_dir.exists() {
+        return Err(anyhow!("Directory {:?} already exists", private_keys_dir));
+    }
+
+    fs::create_dir_all(private_keys_dir)?;
+
+    let wgsk_path = private_keys_dir.join("wgsk");
+    let pqsk_path = private_keys_dir.join("pqsk");
+    let pqpk_path = private_keys_dir.join("pqpk");
+
+    let wgsk = Privkey::generate();
```

Comment:
Unfortunately, I am unable to access the source code of this trait, in order to check if proper zeroization, safe-copying, etc. takes place. If the crate does not take care of this, it will be our responsibility.

Reply:
The wireguard private key is simply a 32 byte random string. We already have an internal crate memsec for dealing with such values.

Reply:
Resolved in ca8a587b6fd07f49ebb4ba6dce831d72e61d415a.